Automattic/jetpack
# Jetpack Sync: Gracefully handle exceptions when calculating checksums

Jetpack Sync: Gracefully handle exceptions when calculating checksums

## Proposed changes:

* `Automattic\Jetpack\Sync\Replicastore\Table_Checksum_Users`: Adds missing `use Exception;`
* `Automattic\Jetpack\Sync\Replicastore`: Updates `checksum_histogram` method to gracefully handle exceptions by adding missing try/catch blocks and returning the corresponding `WP_Error`

**Bonus Performance Imrpovement:** I noticed that we are duplicating the same DB query (related to `get_range_edges`) when calculating buckets so I fixed that too.
**Bonus Performance Imrpovement 2:** No need to calculate buckets when retrieving the histogram in `checksum_histogram` when `$only_range_edges` is `true`. This means that we are only requesting the range edges and not the actual checksum histogram.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- Using your sandbox you will need to trigger checksums/fetch range edges only while throwing Exceptions and confirm that you receive a corresponding WP_Error and the Exception was caught
- Using your sandbox ensure that when triggering checksums the buckets calculated match with/without the PR applies
- You will need to also test this on WPCOM too (see instructions below for testing on Simple sites)
